File: ios/VoiceKitEventEmitter.m
#import <React/RCTBridgeModule.h>
#import <React/RCTEventEmitter.h>

@interface RCT_EXTERN_MODULE(VoiceKitEventEmitter, RCTEventEmitter)

RCT_EXTERN_METHOD(supportedEvents)

+ (BOOL)requiresMainQueueSetup
{
  return NO;
}

@end

File: ios/VoiceKitEventEmitter.swift
import Foundation

@objc(VoiceKitEventEmitter)
class VoiceKitEventEmitter: RCTEventEmitter {
  public static var shared: VoiceKitEventEmitter!

  override init() {
    super.init()
    VoiceKitEventEmitter.shared = self
  }

  override func supportedEvents() -> [String]! {
    ["RNVoiceKit.result", "RNVoiceKit.partial-result", "RNVoiceKit.availability-change", "RNVoiceKit.listening-state-change", "RNVoiceKit.error"]
  }
}
